src/TBuffer.cpp

```diff
@@ -2253,6 +2253,64 @@ void TBuffer::append(const QString& text, int sub_start, int sub_end, TChar form
     }
 }
 
+// Wraps text to max line length of mWrapAt
+// Applies indentation of mWrapIndent to wrapped lines
+QString TBuffer::wrapText(const QString& text) const
+{
+    if (mWrapAt <= mWrapIndent) {
+        qWarning() << "mWrapAt (" << mWrapAt << ") is too small to accommodate mWrapIndent (" << mWrapIndent << ")";
+    }
+
+    const QString wordBreaks = qsl(",.- ");
+    QString wrappedText;
+    QString currentWord;
+    QString currentLine;
+    int wordsInCurrentLine = 0;
+
+    for (int i = 0; i < text.size(); i++) {
+        bool const at_newline = text.at(i) == '\n';
+        if (at_newline) {
+            currentLine += currentWord;
+            wrappedText += '\n' + currentLine;
+            currentLine.clear();
+            currentWord.clear();
+            wordsInCurrentLine = 0;
+            continue;
+        }
+
+        currentWord += text.at(i);
+
+        bool const atWordBreak = wordBreaks.indexOf(text.at(i)) > -1;
+        if (atWordBreak) {
+            // Reached break in word
+            // Add current word to the line and reset for next word
+            currentLine += currentWord;
+            currentWord.clear();
+            wordsInCurrentLine++;
+        }
+
+        int const lineLengthWithWord = currentLine.size() + currentWord.size();
+        bool const needNewLine = lineLengthWithWord >= mWrapAt;
+
+        if (needNewLine) {
+            // Current word would cause an overflow, so wrap the text.
+            if (wordsInCurrentLine == 0) {
```

Comment:
Comparing an integer to zero is more simply put as:
```suggestion
            if (!wordsInCurrentLine) {
```